tracer/tracer.go

```diff
@@ -331,12 +220,14 @@ func NewTracer(ctx context.Context, cfg *Config) (*Tracer, error) {
 		intervals:              cfg.Intervals,
 		hasBatchOperations:     hasBatchOperations,
 		perfEntrypoints:        xsync.NewRWMutex(perfEventList),
-		moduleFileIDs:          moduleFileIDs,
 		reporter:               cfg.Reporter,
 		samplesPerSecond:       cfg.SamplesPerSecond,
 		probabilisticInterval:  cfg.ProbabilisticInterval,
 		probabilisticThreshold: cfg.ProbabilisticThreshold,
-	}, nil
+	}
+	go tracer.pollKobjectClient()
```

Comment:
Let's pass the `context` here to also cleanly exit the goroutine (we wouldn't want to start processing kernel modules if the agent is shutting down).